error: alsaplusplus/pcm.tpp: No such file or directory

Hello sir,
Thnx for sharing in advance.
I built your library with these steps:
`mkdir build`
`cd build`
`cmake ..`
`make`
`sudo make install`
output :
`[100%] Built target AlsaPlusPlus
Install the project...
-- Install configuration: ""
-- Up-to-date: /usr/local/lib/libAlsaPlusPlus.so.0.1.1
-- Up-to-date: /usr/local/lib/libAlsaPlusPlus.so.0
-- Up-to-date: /usr/local/lib/libAlsaPlusPlus.so
-- Up-to-date: /usr/local/include/alsaplusplus/common.hpp
-- Up-to-date: /usr/local/include/alsaplusplus/error.hpp
-- Up-to-date: /usr/local/include/alsaplusplus/mixer.hpp
-- Up-to-date: /usr/local/include/alsaplusplus/pcm.hpp
`
then i created a simple console application with qt creator and added your wrapper to the .pro file.
`LIBS+= -lasound`
`LIBS+= -lalsaplusplus`
on building the example project i face the error which point me to a file which is not there.
`/usr/local/include/alsaplusplus/pcm.hpp:63:12: fatal error: alsaplusplus/pcm.tpp: No such file or directory
   #include <alsaplusplus/pcm.tpp>`

any idea what is wrong.
I use elementary os 5.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -13,7 +13,7 @@ jobs:
           name: Build
           command: |
             mkdir build && cd build
-            cmake ..
+            cmake .. -DWITH_EXAMPLES=ON
             make
       - run:
           name: Install
@@ -34,7 +34,7 @@ jobs:
           name: Build
           command: |
             mkdir build && cd build
-            cmake ..
+            cmake .. -DWITH_EXAMPLES=ON
             make
       - run:
           name: Install
@@ -55,7 +55,7 @@ jobs:
           name: Build
           command: |
             mkdir build && cd build
-            cmake ..
+            cmake .. -DWITH_EXAMPLES=ON
             make
       - run:
           name: Install

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,6 +17,7 @@ set(HEADERS
   ${HEADER_DIR}/alsaplusplus/error.hpp;
   ${HEADER_DIR}/alsaplusplus/mixer.hpp;
   ${HEADER_DIR}/alsaplusplus/pcm.hpp;
+  ${HEADER_DIR}/alsaplusplus/pcm.tpp;
 )
 
 include_directories(${HEADER_DIR})

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,57 +6,65 @@ set(AlsaPlusPlus_VERSION_MINOR 1)
 set(AlsaPlusPlus_VERSION_PATCH 1)
 set(AlsaPlusPlus_VERSION ${AlsaPlusPlus_VERSION_MAJOR}.${AlsaPlusPlus_VERSION_MINOR}.${AlsaPlusPlus_VERSION_PATCH})
 
-option(WITH_SAMPLES "Build and install samples of library usage" OFF)
+option(WITH_EXAMPLES "Build and install example programs" OFF)
 option(INSTALL_HEADERS "Install library headers" ON)
 
 set(CMAKE_CXX_STANDARD 14)
 
 set(HEADER_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
 set(HEADERS
-    ${HEADER_DIR}/alsaplusplus/common.hpp;
-    ${HEADER_DIR}/alsaplusplus/error.hpp;
-    ${HEADER_DIR}/alsaplusplus/mixer.hpp;
-    ${HEADER_DIR}/alsaplusplus/pcm.hpp;)
+  ${HEADER_DIR}/alsaplusplus/common.hpp;
+  ${HEADER_DIR}/alsaplusplus/error.hpp;
+  ${HEADER_DIR}/alsaplusplus/mixer.hpp;
+  ${HEADER_DIR}/alsaplusplus/pcm.hpp;
+)
 
 include_directories(${HEADER_DIR})
 
-add_library(${PROJECT_NAME}
-            SHARED
-            src/error.cpp
-            src/mixer.cpp
-            src/pcm.cpp)
+add_library(
+  ${PROJECT_NAME} SHARED
+  src/error.cpp
+  src/mixer.cpp
+  src/pcm.cpp
+)
 
-set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${AlsaPlusPlus_VERSION} SOVERSION ${AlsaPlusPlus_VERSION_MAJOR})
-target_link_libraries(${PROJECT_NAME}  asound)
+set_target_properties(
+  ${PROJECT_NAME} PROPERTIES
+  VERSION ${AlsaPlusPlus_VERSION}
+  SOVERSION ${AlsaPlusPlus_VERSION_MAJOR}
+)
+target_link_libraries(${PROJECT_NAME} asound)
 
-if(WITH_SAMPLES)
-    add_executable(set_volume
-                   examples/set_volume.cpp)
-    target_link_libraries(set_volume
-                          ${PROJECT_NAME})
+if(WITH_EXAMPLES)
+  add_executable(set_volume examples/set_volume.cpp)
+  target_link_libraries(set_volume ${PROJECT_NAME})
 
-    add_executable(wav_player
-                   examples/wav_player.cpp)
-    target_link_libraries(wav_player
-                          ${PROJECT_NAME})
+  add_executable(wav_player examples/wav_player.cpp)
+  target_link_libraries(wav_player ${PROJECT_NAME})
 
-    install(FILES
-            examples/set_volume.cpp
-            examples/wav_player.cpp
-            DESTINATION
-            share/alsaplusplus/examples)
-endif(WITH_SAMPLES)
+  install(
+    TARGETS set_volume wav_player
+    DESTINATION share/alsaplusplus/examples
+  )
+endif(WITH_EXAMPLES)
 
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib)
 
 if(INSTALL_HEADERS)
-    install(FILES ${HEADERS} DESTINATION include/alsaplusplus)
+  install(
+    FILES ${HEADERS}
+    DESTINATION include/alsaplusplus
+  )
 endif(INSTALL_HEADERS)
 
 if(NOT TARGET uninstall)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
-                 "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
-                 IMMEDIATE @ONLY)
-  add_custom_target(uninstall COMMAND
-                    ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
+  configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
+    IMMEDIATE @ONLY
+  )
+  add_custom_target(
+    uninstall COMMAND
+    ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake
+  )
 endif(NOT TARGET uninstall)

--- a/README.md
+++ b/README.md
@@ -2,8 +2,29 @@
 
 [![CircleCI](https://circleci.com/gh/maximus5684/alsaplusplus.svg?style=svg)](https://circleci.com/gh/maximus5684/alsaplusplus)
 
-A basic C++11 (or higher) wrapper for the ALSA C API on Linux.
+A basic C++14 wrapper for the ALSA C API on Linux.
 
-Requirements:
+## Requirements:
 * A Linux-based system
+* CMake 3.1 or higher
 * The ALSA C API (libasound2-dev on Ubuntu/Debian)
+
+## Installation:
+1. Clone this repository.
+2. In the cloned folder, run:
+
+```
+mkdir build && cd build
+cmake ..
+make
+sudo make install
+```
+
+If you want the example files to be built/installed, change the `cmake` command above to:
+
+```
+cmake .. -DWITH_EXAMPLES=ON
+```
+
+## Usage:
+See the example files in the repository or the header files.

